pandas/core/algorithms.py

```diff
@@ -2149,3 +2149,21 @@ def _sort_tuples(values: np.ndarray[tuple]):
     arrays, _ = to_arrays(values, None)
     indexer = lexsort_indexer(arrays, orders=True)
     return values[indexer]
+
+
+def make_duplicates_of_left_unique_in_right(left, right) -> np.ndarray:
```

Comment:
can you type the input args.

can you add an example here.

is this unit tested?

Reply:
Added unittests, forgot them obviously, and typed the inputs. Example is below